controllers/configuration_controller.go

```diff
@@ -1336,25 +1483,42 @@ func (meta *TFConfigurationMeta) RenderConfiguration(configuration *v1beta2.Conf
 	}
 }
 
-// GetGitCredentialsSecret will get the secret containing the SSH private key & known_hosts
-func GetGitCredentialsSecret(ctx context.Context, k8sClient client.Client, secretRef *v1.SecretReference) (*v1.Secret, error) {
+func GetSecretOrConfigMap(ctx context.Context, k8sClient client.Client, isSecret bool, ref *v1.SecretReference, neededKeys []string, errMsg string, keyErrMsg string) (metav1.Object, error) {
 	secret := &v1.Secret{}
-
-	namespacedName := client.ObjectKey{Name: secretRef.Name, Namespace: secretRef.Namespace}
-	err := k8sClient.Get(ctx, namespacedName, secret)
+	configMap := &v1.ConfigMap{}
+	var err error
+	if isSecret {
+		namespacedName := client.ObjectKey{Name: ref.Name, Namespace: ref.Namespace}
+		err = k8sClient.Get(ctx, namespacedName, secret)
+	} else {
+		namespacedName := client.ObjectKey{Name: ref.Name, Namespace: ref.Namespace}
+		err = k8sClient.Get(ctx, namespacedName, configMap)
+	}
 	if err != nil {
-		errMsg := "Failed to get git credentials secret"
-		klog.ErrorS(err, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
+		klog.ErrorS(err, errMsg, "Name", ref.Name, "Namespace", ref.Namespace)
 		return nil, errors.Wrap(err, errMsg)
 	}
-
-	needSecretKeys := []string{GitCredsKnownHosts, v1.SSHAuthPrivateKey}
-	for _, key := range needSecretKeys {
-		if _, ok := secret.Data[key]; !ok {
-			err := errors.Errorf("'%s' not in git credentials secret", key)
-			return nil, err
+	if len(neededKeys) > 0 {
```

Comment:
No need for this if. If the neededKeys is empty, forloop below will skip itself.

Reply:
removed the unwanted if check.